pilot/pkg/model/push_context.go

```diff
@@ -936,17 +936,30 @@ func (ps *PushContext) getSidecarScope(proxy *Proxy, workloadLabels labels.Colle
 }
 
 // DestinationRule returns a destination rule for a service name in a given domain.
-func (ps *PushContext) DestinationRule(proxy *Proxy, service *Service) *config.Config {
+func (ps *PushContext) DestinationRule(proxy *Proxy, service *Service, direction TrafficDirection) *config.Config {
 	if service == nil {
 		return nil
 	}
-
 	// If proxy has a sidecar scope that is user supplied, then get the destination rules from the sidecar scope
 	// sidecarScope.config is nil if there is no sidecar scope for the namespace
 	if proxy.SidecarScope != nil && proxy.Type == SidecarProxy {
 		// If there is a sidecar scope for this proxy, return the destination rule
 		// from the sidecar scope.
-		return proxy.SidecarScope.destinationRules[service.Hostname]
+		destinationRuleConfig := proxy.SidecarScope.destinationRules[service.Hostname]
```

Comment:
I would expect I can have

DR: google.com, for app=foo
DR: google.com, for app=bar

And both work. However, the `destinationRules` map only holds ONE DR for a given hostname. Because of that, we will probably get `nil` for one of them. Both should be possible. I think in the sidecarscope construction (which is in the context of a single workload) we should do the label selection. At this point we have already lost weight

Reply:
Have made the destionationRules hold a list per host, so that we can do the filtering based on workloadSelector